processor/transformer/transformer.go

```diff
@@ -375,11 +375,22 @@ func (trans *handle) request(ctx context.Context, url, stage string, data []Tran
 	// endless backoff loop, only nil error or panics inside
 	_ = backoff.RetryNotify(
 		func() error {
+			transformationID := ""
+			if len(data[0].Destination.Transformations) > 0 {
```

Comment:
Are you sure at this stage `data` has at least one element?

Reply:
There is a check for that here https://github.com/rudderlabs/rudder-server/pull/5055/files#diff-e5cf9e714ad0653a0fbb461268f1bae1b457735baa48d6658e40ed2840c124f3L361